File: backend/app/schemas/video.py
from datetime import datetime
from pydantic import BaseModel


class VideoResponse(BaseModel):
    id: str
    title: str
    filename: str
    file_size: int
    status: str
    created_at: datetime
    model_config = {"from_attributes": True}


class AnalysisResponse(BaseModel):
    teaching_style: str
    rhythm: str
    guidance_method: str
    core_philosophy: str
    high_freq_words: list[str]
    model_config = {"from_attributes": True}


class FrameResponse(BaseModel):
    id: str
    frame_path: str
    timestamp: str
    frame_type: str
    pose_name: str
    description: str
    model_config = {"from_attributes": True}


class VideoDetailResponse(VideoResponse):
    analysis: AnalysisResponse | None = None
    frames: list[FrameResponse] = []


class GenerateCaptionRequest(BaseModel):
    frame_descriptions: list[str]
    style: str = "轻松愉悦"  # 轻松愉悦 | 专业引导 | 鼓励感恩

File: frontend/src/types/video.ts
export interface VideoAnalysis {
  teaching_style: string;
  rhythm: string;
  guidance_method: string;
  core_philosophy: string;
  high_freq_words: string[];
}

export interface VideoFrame {
  id: string;
  frame_path: string;
  timestamp: string;
  frame_type: string;
  pose_name: string;
  description: string;
}

export interface Video {
  id: string;
  title: string;
  filename: string;
  file_size: number;
  status: string;
  created_at: string;
}

export interface VideoDetail extends Video {
  analysis: VideoAnalysis | null;
  frames: VideoFrame[];
}

export interface GenerateCaptionRequest {
  frame_descriptions: string[];
  style?: string;
}
